src/main/java/com/deavensoft/paketomat/center/PackageServiceImpl.java

```diff
@@ -0,0 +1,109 @@
+package com.deavensoft.paketomat.center;
+
+import com.deavensoft.paketomat.center.model.Package;
+import com.deavensoft.paketomat.center.model.Paid;
+import com.deavensoft.paketomat.center.model.Status;
+import com.deavensoft.paketomat.email.EmailDetails;
+import com.deavensoft.paketomat.email.EmailService;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.time.LocalDateTime;
+
+import java.security.SecureRandom;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class PackageServiceImpl implements PackageService {
+    private final PackageRepository packageRepository;
+    private final EmailService emailService;
+    @Autowired
+
+    public List<Package> getAllPackages() {
+        return packageRepository.findAll();
+    }
+
+    public void save(Package p) {
+        packageRepository.save(p);
+    }
+
+    public Optional<Package> findPackageById(Long id) {
+        return packageRepository.findById(id);
+    }
+
+    public void deletePackageById(Long id) {
+        packageRepository.deleteById(id);
+    }
+
+    @Override
+    public void deleteAll() {
+        packageRepository.deleteAll();
+    }
+
+    @Override
+    public void updateStatus(Long code, Status status) {
+        List<Package> packages = packageRepository.findAll();
+        for (Package p : packages) {
+            if(p.getCode().equals(code)){
+                p.setStatus(status);
+                LocalDateTime date = LocalDateTime.now();
+                p.setDate(date);
+                packageRepository.save(p);
+            }
+        }
+    }
+    public void payment(Long id, Paid paid)
+    {
+        List<Package> packages = packageRepository.findAll();
```

Comment:
You can directly use `packageRepository.findById() `

- [ ] Fix it

Reply:
Fixed.